bodhi/server/consumers/masher.py

```diff
@@ -963,6 +964,40 @@ def sanity_check_repo(self):
 
         return True
 
+    def wait_for_repo_signature(self):
+        """Wait for a repo signature to appear."""
+        notifications.publish(
+            topic="repo.signature.wait",
+            msg=dict(repo=self.id, agent=self.agent, path=self.path),
+            force=True,
+        )
+        if config.get('wait_for_repo_sig'):
+            self.save_state(ComposeState.signing_repo)
+            sigpaths = []
+            repopath = os.path.join(self.path, 'compose', 'Everything')
+            for arch in os.listdir(repopath):
+                if arch == 'source':
+                    sigpaths.append(os.path.join(repopath, arch, 'tree', 'repodata',
+                                                 'repomd.xml.asc'))
+                else:
+                    sigpaths.append(os.path.join(repopath, arch, 'os', 'repodata',
+                                                 'repomd.xml.asc'))
+
+            self.log.info('Waiting for signatures in %s' % sigpaths)
+            while True:
```

Comment:
Is it safe to wait basically "forever" and never fail ? 
I could imagine a few race conditions where this would slowly exhaust ressources, so maybe there should be some limit somewhere, unless that's already handled externally.

Reply:
@mscherer Yeah I had a similar thought - we probably want it to fail if it's not signed after some timeout, just so it doesn't sit around forever.

Reply:
I would say that if there's something wrong to the point where we can't get a repo signed, releng should just abort the push by hand and disable the option.
The problem is that it might be that there's a backlog for signing up to multiple hours, and then there's some time needed for releng to debug it.
If we at some point decide to turn this on and we start shipping configurations with repo_gpgcheck=1, we just cannot ship unsigned repos anymore.

So, I personally think that we should not have an automated timeout, and any timing out should be done by releng.